src/main/java/apoc/convert/Convert.java

```diff
@@ -163,5 +165,85 @@ else if (list.getClass().isArray() && !list.getClass().getComponentType().isPrim
 	public List<Relationship> toRelationshipList(@Name("list") Object list) {
         return convertToList(list, Relationship.class);
 	}
-	
+
+    @UserFunction
+    @Description("apoc.convert.toInteger(value) | tries it's best to convert the value to an integer")
+    public Number toInteger(@Name("object") Object obj) {
+        if (obj == null || obj.equals("")) {
+            return null;
+        }
+
+        Types varType = Types.of(obj);
+        Number result;
+        switch (varType) {
+            case INTEGER:
+            case FLOAT:
+                result = ((Number) obj).intValue();
+                break;
+            case STRING:
+                result = parseIntString((String)obj);
+                break;
+            case BOOLEAN:
+                result = ((boolean) obj) ? 1 : 0;
+                break;
+            default:
+                throw new RuntimeException("Supported types are: Integer, Float, String, Boolean, HexString");
+        }
+        return result;
+    }
+
+    private Integer parseIntString(String input) {
+        if (input.equals("true")) {
+            return 1;
+        } else if (input.equals("false")) {
+            return 0;
+        } else if (input.startsWith("0x")) {
+            return Integer.valueOf(input.substring(2), 16);
+        } else {
+            return (int) Float.parseFloat(input);
```

Comment:
Shouldn't we have separate parsers for Long.parseLong() and (long)Double.parseDouble() based on existence of decimal point?

Reply:
There's a test for that:

'''
testCall(db, "return apoc.convert.toInteger('123') as value", r -> assertEquals(123L, r.get("value")));
        testCall(db, "return apoc.convert.toInteger('123.15') as value", r -> assertEquals(123L, r.get("value")));
'''

and it seems it's not necessary to differentiate based decimal point.